### DISCUSSION TITLE
help understanding log to debug missing libs

### DISCUSSION THREAD

**Discussion by ramarro123** [renovatebot/renovate#24574] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

Placeholder value, please select the correct response from the dropdown

### Wanted end result.

Hello,

i did setup renovate against a private gitlab (self hosted) repository

after your help i managed to add this
  RENOVATE_HOST_RULES: '[{"hostType": "gitlab",  "matchHost": "${CI_SERVER_HOST}", "token": "${GITLAB_ACCESS_TOKEN}"}]'

to enable secure access to repo

and this

        "registryUrls": ["https://artifactrepository.bansel.it/nexus/content/groups/public","https://gitlab.fabrick.dev/api/v4/groups/155/-/packages/maven"]

the one that are hosted  on gitlab repo give
         "https://gitlab.fabrick.dev/api/v4/groups/155/-/packages/maven/it/sella/externals/fabrickcore/fabrickcore-spring-boot-starter-test/maven-metadata.xml (GET,404)": 1,

and a row that look like that

DEBUG: GET https://gitlab.fabrick.dev/api/v4/groups/155/-/packages/maven/com/fabrick/fabricktest/fabricktest/maven-metadata.xml = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=63) (repository=components/fabrick-it/services-provider/fabricktest)

the file is there (if i try access i can see it) so my last idea is that i am not using RENOVATE_HOST_RULES in the correct way.

do you have any hint or clue to help me understanding where i am doing mistakes?

thankyou!

### What you tried so far.

-

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by secustor** [comment]
Probably your auth is failing check the trace logs if the authentification headers are added to the requests.

**Comment by rarkins** [comment, marked as answer]
Try removing the hostType part of your host rule. The log you've shown is for a maven lookup, which would mean hostType=maven not gitlab.

**Reply by ramarro123** [reply]
thankyou it worked!
